src/webgpu/shader/execution/builtin/select.spec.ts

```diff
@@ -133,5 +133,56 @@ Please read the following guidelines before contributing:
 https://github.com/gpuweb/cts/blob/main/docs/plan_autogen.md
 `
   )
-  .params(u => u.combine('placeHolder1', ['placeHolder2', 'placeHolder3']))
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('component', ['b', 'f', 'i', 'u'] as const)
+      .combine('vectorize', [2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const componentType = dataType[t.params.component as scalarKind].type;
+    const cons = dataType[t.params.component as scalarKind].constructor;
+
+    // Exercise all possible bit vectors as the selection parameter.
+    const cases = [];
+    for (let i3 = 0; i3 < 2; i3++) {
+      for (let i2 = 0; i2 < 2; i2++) {
+        for (let i1 = 0; i1 < 2; i1++) {
+          for (let i0 = 0; i0 < 2; i0++) {
+            const k = (i3 << 3) | (i2 << 2) | (i1 << 1) | i0;
+            // The 'a' and 'b' values must have different parity, so that in
+            // the bool case they yield different values.
+            const a0 = cons(k);
+            const b0 = cons(16 * k + 1);
+            cases.push({
+              input: [a0, b0, makeBool(i0)],
+              expected: i0 === 0 ? a0 : b0,
+            });
+            const a1 = cons(k + 1);
+            const b1 = cons(16 * k + 2);
+            cases.push({
+              input: [a1, b1, makeBool(i1)],
+              expected: i1 === 0 ? a1 : b1,
+            });
+            if (t.params.vectorize > 2) {
```

Comment:
So you're expanding the number of cases to match the vectorization width, so the test framework can pack these scalars into a vector. Clever, but it took me a while to get my head around all of this.
For readability, it might be easier to understand if you just constructed the vectors here.
Alternatively, a comment wouldn't go amiss.

Reply:
I've rewritten it to be more explicit and readable.